vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -812,7 +1082,20 @@ private void syncCustomerAttrsData(SDDCSoftwareConfig vcInfo) {
                      String assetName = ipMappings[0].getAssetname();
                      Asset asset = restClient.getAssetByName(assetName).getBody();
                      if (asset != null) {
-                        newMapping.setAsset(asset.getId());
+                        String assetId = asset.getId();
+                        newMapping.setAsset(assetId);
+                        Map<String, Map<String, Map<String, String>>> metricsFormulars =
+                              asset.getMetricsformulars();
+                        if (null != metricsFormulars) {
+                           Map<String, Map<String, String>> metrics =
+                                 metricsFormulars.get(FlowgateConstant.HOST_METRICS);
+                           if (metrics == null || metrics.isEmpty()) {
+                              metrics = new HashMap<>();
+                              feedAssetMetricsFormulars(metrics, assetId);
+                              metricsFormulars.put(FlowgateConstant.HOST_METRICS, metrics);
+                              restClient.saveAssets(asset);
+                           }
+                        }
```

Comment:
refactor into a method . Please also check when the mapping already exist situation.